workflows/scripts/consensus.py

```diff
@@ -105,11 +106,30 @@ def build_consensus(
     output = str(snakemake.output)
     config_path = str(snakemake.params["consensus_configs"])
     log_path = str(snakemake.log)
-
-    # concatenated all Normalized aggregated profiles
-    build_consensus(
-        profile_list=inputs,
-        consensus_file_out=output,
-        config=config_path,
-        log_file=log_path,
-    )
+    enable_profiling = snakemake.config["enable_profiling"]
+
+    # normalization step
+    if enable_profiling:
+        # setting up output path to benchmark folder
+        benchmark_dir = pathlib.Path(str(snakemake.config["benchmark_dir"])).resolve(
+            strict=True
+        )
+        output_path = benchmark_dir / "consensus_benchmark.bin"
+
+        # anything below this context manager will be profiled
+        with memray.Tracker(output_path):
+            # concatenated all Normalized aggregated profiles
+            build_consensus(
+                profile_list=inputs,
+                consensus_file_out=output,
+                config=config_path,
+                log_file=log_path,
+            )
+    else:
+        # concatenated all Normalized aggregated profiles
+        build_consensus(
+            profile_list=inputs,
+            consensus_file_out=output,
+            config=config_path,
+            log_file=log_path,
+        )
```

Comment:
This comment applies to the pattern of this code block across many of the `workflows/scripts` files. Consider adding a utility function which can help perform some of this work to avoid the repetition of these edits. For example, could the benchmark_dir be resolved and a `bin` path created for each of these with a function? Similarly, consider a custom decorator function help run the same workflows with/without `enable_profiling` (to avoid the duplicate calls to, for example, `build_consensus` with differing context).

Reply:
Thanks for pointing this out. I will make this an issue. I have yet figured out a way to reduce code duplication caused by the profiling approach. If you have any resources that will help me achieve that, that will be greatly appreciated.